apis/v1/gateway_types.go

```diff
@@ -295,6 +295,24 @@ type GatewaySpec struct {
 	//
 	// +optional
 	AllowedListeners *AllowedListeners `json:"allowedListeners,omitempty"`
+
+	// TLSConfigs stores TLS configurations for a Gateway.
+	//
+	//   - If the `port` field in `TLSConfig` is not set, the TLS configuration applies
+	//     to all listeners in the gateway. We call this `default` configuration.
+	//   - If the `port` field in `TLSConfig` is set, the TLS configuration applies
+	//     only to listeners with a matching port. Each port requires a unique TLS configuration.
+	//   - Per-port configurations can override the `default` configuration.
+	//   - The `default` configuration is optional. Clients can apply TLS configuration
+	//     to a subset of listeners by creating only per-port configurations.
+	//     Listeners with a port that does not match any TLS configuration will
+	//     not have `frontendValidation` set.
+	//
+	// Support: Core
+	// +optional
+	//
+	// <gateway:experimental>
+	TLSConfigs []TLSConfig `json:"tlsConfigs,omitempty"`
```

Comment:
🤔 This field name feels a little funny to me. Something about `configs` as a plural word doesn't seem quite right.  Some alternatives that may not be better:

1. `tlsByPort`
2. `tlsPerPort`
3. `tls` 
4. `frontendTLS`

Since even from the outset `port` is optional, 1 or 2 probably aren't great. Since there's at least a theoretical world where we add something TLS related here that isn't tied to frontend TLS, we could regret 4 in the future. So maybe 3 is the least bad, although even that adds some confusion as that `tls` field would have no overlap with [listener.tls](https://github.com/kubernetes-sigs/gateway-api/blob/78496d8f897031a5335fb1fbcdfe422721646f4b/apis/v1/gateway_types.go#L426). Open to other ideas here.

/cc @arkodg @youngnick

Reply:
if there is no strict rule for pluralizing / `s` slice fields, then +1 for `tls`

Reply:
I added a different proposal above, but this could also work.

Reply:
After some reflection, I think that @kl52752's option here is better, agreed. Separating the default config into its own section lets us say "the default config applies to all listeners with matching ports that terminate TLS, unless there is a matching config in the `tlsPerPort` section. In that case only the config in the `tlsPerPort` section applies."